packages/malloy/src/lang/ast/ast-types.ts

```diff
@@ -199,6 +199,10 @@ function term(f: Fragment[]): Fragment[] {
   return f;
 }
 
+function wrapInParens(f: Fragment[]): Fragment[] {
+  return ["(", ...f, ")"];
+}
```

Comment:
I considered only wrap if not currently wrapped, but I'm not sure if it's safe:

```ts
function wrapInParens(f: Fragment[]): Fragment[] {
  if (f.length && f[0] === '(' && f[f.length] === ')') {
    return f;
  }
  return ["(", ...f, ")"];
}
```

Reply:
Yeah, it isn't safe. There is no guarantee that a fragment doesn't look like

```
[ "(  'start of some string which contains)" ]
```

or

```
[ "(expr1) or (expr2)" ]
```

If fragments were type-protected and we could know that they were well formed, then something like this check would work.